docs/hydra/guides/oauth2-webhooks.mdx

```diff
@@ -0,0 +1,295 @@
+---
+id: claims-at-refresh
+title: OAuth2 webhooks
+sidebar_label: OAuth2 webhooks
+keywords:
+  - webhooks
+  - custom
+  - customize
+  - claims
+  - jwt
+  - access
+  - token
+---
+
+# Customizing claims with OAuth2 webhooks
+
+```mdx-code-block
+import CodeBlock from '@theme/CodeBlock';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem'
+```
+
+You can modify aspects of the OpenID Connect and access tokens returned from Hydra's OAuth2 token endpoint. A typical use case is
+adding custom claims to the tokens issued by Hydra.
+
+To customize the token response, you must configure a webhook endpoint in your OAuth2 configuration. Before the token is issued to
+the client, Hydra will call your HTTPS endpoint with information about the OAuth client requesting the token.
+
+For the Authorization Code flow, information about the resource owner's session are also included in the webhook payload.
+
+Your endpoint's response to the webhook will be used to customize the token that Hydra issues to the OAuth client, and optionally
+overwrite the session data stored for the resource owner.
+
+Using webhooks is supported for all grant types (flows).
```

Comment:
```suggestion
Using webhooks is supported for all grant types (flows).
```

which one is it? probably good to be exact here, easy to get confused with OAuth2 terminology IMO.

Reply:
I've regarded flow and grant type as synonymous.

Reply:
Gotcha. Maybe not super relevant for this PR but IMO we should pick one and stick with it in the documentation in general? 

It can be confusing if we sometimes talk about flows and in other places talk about grant types while meaning the same thing.

Reply:
Yes, I agree. Calling that out of scope for this PR tho.